README.md

```diff
@@ -0,0 +1,153 @@
+
+**Optical Dataset:**
+https://drive.google.com/drive/folders/1UHE8pRFGZkHiOwmwc8tFI77c620U59eD
+
+**SAR Dataset:**
+https://drive.google.com/drive/folders/1WvKc7AxSiFknl1EuICnlLY2hU487qtB-
+
+### Dataset Organization
+After downloading, organize your datasets in the `data` folder as follows:
+
+```
+data/
+├── flood_optical/
+│   ├── train/
+│   │   ├── images/
+│   │   └── masks/
+│   └── test/
+│       ├── images/
+│       └── masks/
+└── flood_sar/
+    ├── train/
+    │   ├── images/
+    │   └── masks/
+    └── test/
+        ├── images/
+        └── masks/
+```
+
+### Combined Dataset Setup
+For combined training using both optical and SAR data, simply merge the datasets into a single folder:
+
+```
+data/
+└── flood_combined/
+    ├── train/
+    │   ├── images/  # Contains both optical and SAR images
+    │   └── masks/   # Contains corresponding masks
+    └── test/
+        ├── images/  # Contains both optical and SAR images
+        └── masks/   # Contains corresponding masks
+```
+
+**Note**: The combined dataset is created by copying all images and masks from both `flood_optical` and `flood_sar` datasets into a single `flood_combined` folder. The model will automatically process both data types during training.
+
+## Running the Demos
+
+**Note**: The PyTorch implementation is more developed and feature-complete compared to the JAX version. The results were obtained from the PyTorch implmentation.
+
+### Centralized Training (PyTorch)
+
+#### Classical U-Net
+```bash
+python centralized_demoTorchClassical.py
+```
+
+#### Quantum U-Net (PennyLane)
+```bash
+python centralized_demoTorchQ.py
+```
+
+### Federated Learning (PyTorch)
+
+#### Classical U-Net Federated
+```bash
+python federated_demoTorchClassical.py
+```
+
+#### Quantum U-Net Federated
+```bash
+python federated_demoTorchQ.py
+```
+
+### JAX Implementation (Experimental)
+
+#### Centralized JAX
+```bash
+python centralized_demoJAX.py
+```
+
+#### Federated JAX
+```bash
+python federated_demoJAX.py
+```
+
+## Configuration
+
+All demos use configuration dictionaries that can be modified in the respective files:
+
+- `image_size`: Input image resolution (default: 128)
+- `base_channels`: Base number of channels (default: 32)
+- `quantum_channels`: Number of channels processed by quantum circuits (default: 32)
+- `batch_size`: Training batch size (default: 8)
+- `learning_rate`: Learning rate (default: 1e-4)
+- `quantum_backend`: 'pennylane' or 'ptlayer' (PyTorch only)
+
+## Output
+
+Results are saved in the `results/` directory with the following structure:
+- Training curves (loss, accuracy, AUC)
+- Model checkpoints
+- Sample visualizations
+- Evaluation metrics
+
+## Quantum Backends
+
+- **PennyLane**: Quantum machine learning library with various simulators
+- **PTLayer**: Photonic quantum computing backend (requires Orca setup)
+
+## Related Work
+
+Our QVUNet architectures are inspired by the quantum hybrid diffusion models proposed in:
+
+```bibtex
```

Comment:
We need to add more references to our previous work. In particular:

https://iopscience.iop.org/article/10.1088/2632-2153/ad8c11
https://pubs.acs.org/doi/10.1021/acs.iecr.4c03805 (and its corresponding repository)
https://openreview.net/forum?id=Is6AZWd0g5&referrer=%5Bthe%20profile%20of%20David%20E.%20Bernal%20Neira%5D(%2Fprofile%3Fid%3D~David_E._Bernal_Neira1)